pkg/status/types.go

```diff
@@ -5,13 +5,35 @@
 
 package status
 
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+)
+
 // CLCChecks is used to unmarshall the runner expvar payload for CLC Runner
 type CLCChecks struct {
 	Checks map[string]map[string]CLCStats `json:"Checks"`
 }
 
 // CLCStats is used to unmarshall the stats needed from the runner expvar payload
 type CLCStats struct {
-	AverageExecutionTime int `json:"AverageExecutionTime"`
-	MetricSamples        int `json:"MetricSamples"`
+	AverageExecutionTime int  `json:"AverageExecutionTime"`
+	MetricSamples        int  `json:"MetricSamples"`
+	LastExecFailed       bool `json:"LastExecFailed"`
+}
+
+// UnmarshalJSON overwrite the unmarshall method
+func (d *CLCStats) UnmarshalJSON(data []byte) error {
+	var stats check.Stats
+	if err := json.Unmarshal(data, &stats); err != nil {
+		return err
+	}
+	d.AverageExecutionTime = int(stats.AverageExecutionTime)
+	d.MetricSamples = int(stats.MetricSamples)
+	if stats.LastError != "" {
```

Comment:
Were we not surfacing the error before ?

Reply:
This is used here: https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/metadata/agentchecks/agentchecks.go#L31-L34

Reply:
no, we didn't know in the cluster-agent that a cluster-check has failed.
So in case of Timeout for example, the `AverageExecutionTime ` was high but didn't reflect the business

Reply:
gotcha, that makes sense.